### ISSUE TITLE
accumulator-server.py is not working even inside virtualenv

### ISSUE THREAD

**Issue by WyRe** [telefonicaid/fiware-orion#3570]
Well, I was following your [getting started guide](https://fiware-orion.readthedocs.io/en/master/user/walkthrough_apiv2/index.html#starting-accumulator-server) to turn on the accumulator server and receive Orion notifications in there. I've even created a `virtualenv` like you suggest and installed `Flask==1.0.2` and `pyOpenSSL==19.0.0` but it doesn't work even so. 

First I'm getting a syntax error in lin 64 about a `print` function, then if I fix the line as `python` suggest, I get another syntax error at line 76. 

What do you think? Perhaps I need a earlier `python` version?

![Captura de pantalla_2019-11-04_19-37-09](https://user-images.githubusercontent.com/4059108/68147620-b87e9400-ff3a-11e9-892d-6b2ec34ed627.png)

**Comment by WyRe** [comment]
I've tried with `python2` and apparently is working, I would suggest you to make the explicit appointment [in there](https://fiware-orion.readthedocs.io/en/master/user/walkthrough_apiv2/index.html#starting-accumulator-server) (**`python2` version is needed**; I mean). 

Otherwise when that's not explicitly said, I think the most usual thought is the script is designed for `python3`.

I'll keep open this issue because you might want to modify your docu.

**Comment by fgalan** [comment]
Thanks for the feedback! Documentation has been modified in the following PR: https://github.com/telefonicaid/fiware-orion/pull/3572. It would be great if you could have a look and provide feedback :)

In addition, it would be great if somebody could provide a version of accumulator-server.py for python3. It should be just a matter of adapting little things, e.g. as `print "..."` to `print("...")` and things like that.

---------------------

With regards to the problem you have, the following comes to my attention:

![imagen](https://user-images.githubusercontent.com/1534240/68298219-1d500080-0099-11ea-8d64-d5727c052577.png)

Maybe is some IPv4-IPv6 interoperability problem. In theory, Orion is able to send notifications using IPv6 but maybe some issue in the name resolution system or some other obscure OS-related reason is causing problems...

**Comment by WyRe** [comment]
> Maybe is some IPv4-IPv6 interoperability problem. In theory, Orion is able to send notifications using IPv6 but maybe some issue in the name resolution system or some other obscure OS-related reason is causing problems...

I thought exactly the same, that's why I also tried 

```
./accumulator-server.py --port 1028 --url /accumulate --host localhost --pretty-print -v
```

With the same result unfortunately. Regarding to _some other obscure OS-related reason is causing problems..._ I would be fine if someone could reproduce the same issue, Otherwise ... I would point to my system. But I've tried in my two personal computers, laptop and desktop and both are showing the same behaviour; although both have also the same OS (Arch Linux).

---

> In addition, it would be great if somebody could provide a version of accumulator-server.py for python3. It should be just a matter of adapting little things, e.g. as print "..." to print("...") and things like that.

Of course, I'd love to do that if I had enough free time, hehe. It would be a nice way to learn python :+1:

**Comment by WyRe** [comment]
@fgalan I've been thinking about that `debug mode` too:

![Captura de pantalla_2019-11-08_14-02-07](https://user-images.githubusercontent.com/4059108/68478578-8ae26500-0230-11ea-9bff-f8a0eda49229.png)

Is it related with the [line 425](https://github.com/telefonicaid/fiware-orion/blob/master/scripts/accumulator-server.py#L425) in the script?

**Comment by fgalan** [comment]
> Could be helpful to enable that debug mode in order to locate where is the issue?

I think it wouldn't hurt. Enable it and let's see if it brings some light to the problem.

In addition, I have remembered that I had some problems in the past with Flask (the underlying web server in which accumulator-server.py is based) and IPv6. Maybe it has some relation, although it happened many years ago and I don't remember exactly. In the case it can help, this is the link: https://stackoverflow.com/questions/21673068/dual-ipv4-and-ipv6-support-in-flask-applications

**Comment by fgalan** [comment]
> I'm also considering your suggestion to port the code to python3. What's the work flow.

What do you mean by work flow?

**Comment by WyRe** [comment]
> > I'm also considering your suggestion to port the code to python3. What's the work flow.
> 
> What do you mean by work flow?

I mean, should I fork `fiware-orion` then make my own branch, the accumulator port and a pr?

**Comment by fgalan** [comment]
> I mean, should I fork fiware-orion then make my own branch, the accumulator port and a pr?

Yes, that's the usual workflow. You can do it in a new file (e.g. accumulator-server-python3.py) so we can have both versions for a while. Thanks!

**Comment by fernandorr21** [comment]
Sorry, I have the same problem. Could you resolve it?

**Comment by fgalan** [comment]
> Sorry, I have the same problem. Could you resolve it?

Not sure if @jfernandz finally adapted the accumulator to Python 3. Could you tell us, please? Thanks!

**Comment by jfernandz** [comment]
@fgalan I'd like to do it, but for now I'm improving my python knowledge. I think it is not a priority because it is just a simple script for testing purposes. But of course I hope I can port the code soon.

**Comment by vijapandey** [comment]
Issue is still persist . The Server accumulator-server.py is not showing message as per documentation , if any one has resolved the issue ,then pls share the improvement or solution

**Comment by jfernandz** [comment]
@vijapandey lemme check it :wink: I think I finally found a break to work on it :blush:

**Comment by vijapandey** [comment]
> 

Issue has been resolved now . please see above solution

**Comment by jfernandz** [comment]
@vijapandey sure, anyway I was trying to port the script to python3 :relaxed: 

https://github.com/jfernandz/fiware-orion/blob/port/3570_accumulator/scripts/accumulator-server3.py
